tests/testImageDecorrelation.py

```diff
@@ -0,0 +1,253 @@
+        bbox = afwGeom.Box2I(afwGeom.Point2I(0, 0), afwGeom.Point2I(self.im1.shape[0]-1, self.im1.shape[1]-1))
+
+        # All this code to convert the template image array/psf array into an exposure.
+        self.im1ex = afwImage.ExposureF(bbox)
+        self.im1ex.getMaskedImage().getImage().getArray()[:, :] = self.im1
+        self.im1ex.getMaskedImage().getVariance().getArray()[:, :] = self.sig1**2.
+        psfBox = afwGeom.Box2I(afwGeom.Point2I(-20, -20), afwGeom.Point2I(20, 20))
+        psf = afwImage.ImageD(psfBox)
+        psfBox.shift(afwGeom.Extent2I(256, 256))
+        im1_psf_sub = self.im1_psf[psfBox.getMinX():psfBox.getMaxX()+1, psfBox.getMinY():psfBox.getMaxY()+1]
+        psf.getArray()[:, :] = im1_psf_sub
+        psfK = afwMath.FixedKernel(psf)
+        psfNew = measAlg.KernelPsf(psfK)
+        self.im1ex.setPsf(psfNew)
+
+        # Now for the 2nd (science) image
+        self.im2ex = afwImage.ExposureF(bbox)
+        self.im2ex.getMaskedImage().getImage().getArray()[:, :] = self.im2
+        self.im2ex.getMaskedImage().getVariance().getArray()[:, :] = self.sig2**2.
+        psf = afwImage.ImageD(psfBox)
+        im2_psf_sub = self.im2_psf[psfBox.getMinX():psfBox.getMaxX()+1, psfBox.getMinY():psfBox.getMaxY()+1]
+        psf.getArray()[:, :] = im2_psf_sub
+        psfK = afwMath.FixedKernel(psf)
+        psfNew = measAlg.KernelPsf(psfK)
+        self.im2ex.setPsf(psfNew)
+
+    def tearDown(self):
+        del self.im1
+        del self.im1_psf
+        del self.im2
+        del self.im2_psf
+        del self.im1ex
+        del self.im2ex
+
+    def testDiffimCorrection(self):
+        """! Check that the variance of the corrected diffim matches the theoretical value
+        (to within a 1% tolerance).
+        """
+
+        # Create the matching kernel. We use gaussians, so it's just:
+        sig_match = np.sqrt((self.psf2**2. - self.psf1**2.))
+        matchingKernel = measAlg.SingleGaussianPsf(25, 25, sig_match)
+
+        # Create the matched template by convolving the template with the matchingKernel
+        from scipy.ndimage.filters import convolve
+        matched_im1 = convolve(self.im1, matchingKernel.computeImage().getArray(), mode='constant')
+        matched_im1ex = self.im1ex.clone()
+        matched_im1ex.getMaskedImage().getImage().getArray()[:, :] = matched_im1
+
+        # Expected (ideal) variance of difference image
+        expected_var = self.sig1**2 + self.sig2**2
+        print('Expected variance:', expected_var)
+
+        #print(self.im1.var())
+        #print(self.im2.var())
+        #print((self.im1 - self.im2).var())
+        print((matched_im1 - self.im2).var())
+        # Uncorrected diffim - variance is wrong (too low)
+        self.assertNotClose((matched_im1 - self.im2).var(), expected_var, rtol=0.1)
+        # In fact, it should have a variance close to the variance of the science image.
+        self.assertClose((matched_im1 - self.im2).var(), self.sig2**2., rtol=0.1)
+
```

Comment:
assertImagesNearlyEqual will handle the variance plane (since it is just an image). I strongly recommend using it, since it is our standard way to compare images, variances, masks and masked images.